nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -68,11 +75,19 @@ sub runCommand {
     elsif ($arg eq "--flake") {
         $flake = 1;
     }
+    elsif ($arg eq "--kernel") {
+        $n++;
+        $kernel = $ARGV[$n];
+        die "$0: ‘--kernel’ requires an argument\n" unless defined $kernel;
```

Comment:
Maybe you should also handle the case where $kernel starts with -- ?
```suggestion
        die "$0: ‘--kernel’ requires an argument\n" unless defined $kernel && $kernel !~ /^--/;
```

Reply:
@marcusramberg That's (effectively) already handled below, where the value of the `--kernel` argument is checked for validity. I would think the error `invalid kernel: '--root'` would be clear enough